tutorials/sensor/lidar_node.cc

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <ignition/msgs/twist.pb.h>
+#include <ignition/msgs/laserscan.pb.h>
+#include <ignition/transport/Node.hh>
+
+//////////////////////////////////////////////////
+/// \brief Function called each time a topic update is received.
+void cb(const ignition::msgs::LaserScan &_msg)
+{
+  std::string topic_pub = "/cmd_vel";   //publish to this topic
+  ignition::transport::Node node;
+  auto pub = node.Advertise<ignition::msgs::Twist>(topic_pub);
```

Comment:
I initialized the node here and also in the `main`. which I think isn't a good practice at all

Reply:
Oh ok, I understand the issue now. Yeah you're right that it's not good to have this node and publisher recreated every time a new message is received.

I think the way you had it before was better. Sorry for the back-and-forth :bowing_woman:

Reply:
No problem at all